### PR TITLE
Add package onefuzz

### PR DESCRIPTION
**Pull request by cswindle**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Adding a new package named onefuzz

###### Change Log  

- Adding new package onefuzz

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local build
- Buddy Build: [PR-5861](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=396051&view=results) [❌ pipeline fails]

@mfrw on behalf of @cswindle; (reason - may not have permissions):
- Upload to Blobstore: [Pipeline-Link](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=396050&view=results)
- Trigger Test Build: [PR-5861](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=396051&view=results)

### CODE REVIEW

**Comment by cswindle** [comment]
Note that "Spec Linting" seems to be a false error as it wants the changelog removed, but if it is removed then other CI stages fail.

File: SPECS/onefuzz/onefuzz.spec [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+%description
+Project OneFuzz enables continuous developer-driven fuzzing to proactively
+harden software prior to release. With a single command, which can be baked
+into CICD, developers can launch fuzz jobs from a few virtual machines to
+thousands of cores.
+
+%prep
+%setup -q
+%patch0 -p1
```
Comment (mfrw) [inline SPECS/onefuzz/onefuzz.spec:31]: I think it would be cleaner to leverage autosetup:
```suggestion
%autosetup -p1
```

**Comment by cswindle** [comment]
Note that I have raised https://github.com/microsoft/onefuzz/issues/3323 to cover the fact that the source is not able to be built from a tarball without a patch.

**Comment by cswindle** [comment]
Please could you re-run the buddy pipeline for me, as I don't believe I have permission

**Comment by mfrw** [comment]
> Please could you re-run the buddy pipeline for me, as I don't believe I have permission

Test Build v2 : [PR-5861](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=396506&view=results)

File: SPECS/onefuzz/onefuzz.spec [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+harden software prior to release. With a single command, which can be baked
+into CICD, developers can launch fuzz jobs from a few virtual machines to
+thousands of cores.
+
+%prep
+%autosetup -p1
+
+%build
+cd src/agent
+ONEFUZZ_SET_VERSION=%{version} RUSTFLAGS="-llzma" cargo build --release
```
Comment (mfrw) [inline SPECS/onefuzz/onefuzz.spec:34]: @cswindle
On further inspection, there is an edge case which we are hitting here as per the design of AzureLinux build system.
- The toolkit does not allow internet access during a normal build; but how did the local build succeed you might ask; Well, if we build with RUN_CHECK=y (which is only used for tests) Internet access is enabled.

Maybe we could sync-up offline for further discussions. I did manage to build it locally (with RUN_CHECK=y [internet enabled]), but for offline building we might want to vendor sources by doing `cargo vendor` and caching the entire sources that it has to download from the internet.

I did some slight modifications to the BuildRequires, 
```
- BuildRequires: kernel-headers
+ BuildRequires: perl(FindBin)
+ BuildRequires: perl(lib)
```

**Pull request closed without merging**